src/common/GraphicalInformationSet.cpp

```diff
@@ -112,17 +118,44 @@ gsoap_eml2_2::resqml2__GraphicalInformationForWholeObject* GraphicalInformationS
 	return nullptr;
 }
 
-gsoap_eml2_2::resqml2__HsvColor* GraphicalInformationSet::getDefaultColor(common::AbstractObject* targetObject) const
+resqml2__HsvColor* GraphicalInformationSet::getDefaultColor(AbstractObject const* targetObject) const
 {
-	gsoap_eml2_2::resqml2__GraphicalInformationForWholeObject* result = getDefaultGraphicalInformation(targetObject);
+	if (targetObject == nullptr) {
+		throw invalid_argument("The target object cannot be null");
+	}
+
+	resqml2__GraphicalInformationForWholeObject const* result = getDefaultGraphicalInformation(targetObject);
+
 	return result != nullptr && result->ConstantColor != nullptr ? result->ConstantColor : nullptr;
 }
 
-bool GraphicalInformationSet::hasGraphicalInformation(const common::AbstractObject* targetObject) const
+resqml2__ColorInformation* GraphicalInformationSet::getColorInformation(AbstractObject const* targetObject) const
 {
+	if (targetObject == nullptr) {
+		throw invalid_argument("The target object cannot be null");
+	}
+
+	_eml22__GraphicalInformationSet const* gis = static_cast<_eml22__GraphicalInformationSet*>(gsoapProxy2_2);
+
+	for (size_t giIndex = 0; giIndex < gis->GraphicalInformation.size(); ++giIndex) {
+		if (getTargetObjectUuid(giIndex).compare(targetObject->getUuid()) == 0 &&
+			gis->GraphicalInformation[giIndex]->soap_type() == SOAP_TYPE_gsoap_eml2_2_resqml2__ColorInformation) {
+			return static_cast<resqml2__ColorInformation*>(gis->GraphicalInformation[giIndex]);
+		}
+	}
+
+	return nullptr;
+}
+
+bool GraphicalInformationSet::hasDirectGraphicalInformation(const AbstractObject const* targetObject) const
```

Comment:
the const pointer is unuseful since is is a copy.

Reply:
Done